Lib/_collections_abc.py

```diff
@@ -106,11 +106,11 @@ async def _ag(): yield
 ### ONE-TRICK PONIES ###
 
 def _check_methods(C, *methods):
-    mro = C.__mro__
+    mro_dicts = [B.__dict__ for B in C.__mro__]
```

Comment:
Maybe we should use a tuple instead of a list to reduce memory usage, since this collection isn’t modified:


```suggestion
    mro_dicts = tuple(B.__dict__ for B in C.__mro__)
```

Reply:
would it be reasonable to cache this tuple for re-use?

Reply:
But the caching here would need to be more sophisticated, since when the class is updated, the result must also change. Standard caching only takes the input into account, so in this case the input would be cached, but the output should actually differ depending on the class state.

Reply:
can a class mro change after the fact? 

I know the contents of __dict__ could change but if the cache is just holding references to each class' dict itself, then any modifications of any dict should already be reflected in the cache

If the mro can change this becomes invalid

Reply:
to be clear I'm *not* suggesting to cache the result of this function, only the mro dict tuple

Reply:
Ah, got it. I think caching the MRO dict tuple should be safe.

Reply:
Here you also need to decide what exactly we want to optimize, since a tuple gives about ~10% better memory efficiency, while a list is about ~10% faster when iterating in loops.

Reply:
interesting, I didn't know the lists are faster to iterate thru, even with the length check at each iteration. TIL, thanks

In that case I'd opt for the list for top speed in the subclass check but I'm not sure how that fits into the best practices here or what the people want

Reply:
It's not worth debating tiny differences like this until we've verified that the change itself is worth pursing.